www/assets/js/account.js

```diff
@@ -1,150 +1,152 @@
-var $passwordResetButton = document.querySelector('.js-reset-toggle')
-var $passwordReset = document.querySelector('.js-password-reset')
-var $signinForm = document.querySelector('.js-signin-form')
-var $signupForm = document.querySelector('.js-signup-form')
-var $accountForms = document.querySelector('.js-account-forms')
-var $signedinMessage = document.querySelector('.js-signedin-message')
-var $signupButton = document.querySelector('.js-signup-btn')
-var $signoutButton = document.querySelector('.js-signout-btn')
-var $signinToggle = document.querySelector('.js-signin-toggle')
-var $signupToggle = document.querySelector('.js-signup-toggle')
-var $usernameToggle = document.querySelector('.js-username-toggle')
-var $usernameForm = document.querySelector('.js-username-form')
-var $passwordToggle = document.querySelector('.js-password-toggle')
-var $passwordForm = document.querySelector('.js-password-form')
-var $deleteToggle = document.querySelector('.js-delete-toggle')
-
-$passwordResetButton.addEventListener('click', function (event) {
-  event.preventDefault()
+/**
+ * Find all elements with a data-show attribute and apply a click handler
+ * in which we check if an element with a data-action attribute was clicked.
+ * If yes, then set the data-show attribute of the container element to what
+ * is in the data-action of the clicked attribute (after the show- prefix)
+ */
+;[].forEach.call(document.querySelectorAll('[data-show]'), function (el) {
+  el.addEventListener('click', function (event) {
+    var action = event.target.dataset.action
+    var $formsContainer = event.currentTarget
+
+    if (!action) {
+      return
+    }
+
+    event.preventDefault()
+    $formsContainer.dataset.show = action.substr('show-'.length)
+  })
+})
 
-  $passwordReset.classList.remove('hide')
-  document.querySelector('.js-account-login').classList.add('hide')
+/**
+ * Handle signpu form submit
+ */
+document.querySelector('form.signup').addEventListener('submit', function (event) {
+  event.preventDefault()
 
-  var email = document.findElementById('email-reset').value
+  var email = this.querySelector('[name=email]').value
+  var password = this.querySelector('[name=password]').value
 
-  hoodie.account.request({
-    type: 'passwordreset',
-    username: email
+  hoodie.account.signUp({
+    username: email,
+    password: password
   })
 
   .then(function () {
-    alert('done')
+    return hoodie.account.signIn({
+      username: email,
+      password: password
+    })
   })
 
-  .catch(function (error) {
-    alert(error)
-  })
+  .catch(handleError)
 })
 
-$signinForm.addEventListener('submit', function (event) {
+/**
+ * Handle signin form submit
+ */
+document.querySelector('form.signin').addEventListener('submit', function (event) {
```

Comment:
see above. I know the reasons, it’s just not a good reasons for this use case. There will be no refactoring :) It’s all about make it as simple to understand who looks at this for the first time